src/main/java/uk/gov/gchq/palisade/util/UriBuilder.java

```diff
@@ -103,6 +108,7 @@ private IScheme build() {
                         thisFrag
                 );
             } catch (URISyntaxException e) {
+                LOGGER.error("IScheme create threw an error when building a URI ", e);
                 throw new IllegalArgumentException(String.format("Invalid URI: %s:%s%s%s%s", thisScheme, thisAuth, thisPath, "?" + thisQuery, "#" + thisFrag), e);

```

Comment:
This creates another SQ smell of log or throw exception not both. Suggest just throwing the exception

Reply:
done